src/state/slices/assetsSlice/selectors.ts

```diff
@@ -18,15 +19,24 @@ export const selectAssetById = createCachedSelector(
   (byId, assetId) => byId[assetId] || undefined,
 )((_state: ReduxState, assetId: AssetId | undefined): AssetId => assetId ?? 'undefined')
 
-export const selectRelatedAssetIds = createSelector(
+// selects all related assetIds, inclusive of the asset being queried
+export const selectRelatedAssetIdsInclusive = createDeepEqualOutputSelector(
   (state: ReduxState) => state.assets.relatedAssetIndex,
   selectAssetById,
   (relatedAssetIndex, asset): AssetId[] => {
-    const relatedAssetKey = asset?.relatedAssetKey
-    if (!relatedAssetKey) return []
-    return [relatedAssetKey]
-      .concat(relatedAssetIndex[relatedAssetKey] ?? [])
-      .filter(assetId => assetId !== asset.assetId)
+    if (!asset) return []
+    const relatedAssetKey = asset.relatedAssetKey
+    if (!relatedAssetKey) return [asset.assetId]
+    return [relatedAssetKey].concat(relatedAssetIndex[relatedAssetKey] ?? [])
```

Comment:
Could consume `selectRelatedAssetIds` as input selector here and concat this here instead to keep things DRY

Reply:
no ser, `selectRelatedAssetIds` consumes this selector